### PR TITLE
Disable mlir in default builds for now

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Comment by pfultz2** [comment]
So I disabled it by default for dev builds since MLIR fails on other OSs like sles. However, I still enable it when running tidy, so we shouldn't need new tidy fixes when we re-enable it.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -55,6 +55,7 @@ jobs:
           -DMIGRAPHX_ENABLE_GPU=On \
           -DMIGRAPHX_ENABLE_CPU=On \
           -DMIGRAPHX_ENABLE_FPGA=On \
+          -DMIGRAPHX_ENABLE_MLIR=On \
```
Comment (TedThemistokleous) [inline .github/workflows/ci.yaml:58]: Should this be off if we're disabling mlir?

**Review by TedThemistokleous — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (pfultz2) [reply .github/workflows/ci.yaml:58]: No, I want tidy to be enabled so we can still catch tidy issues.

Comment (TedThemistokleous) [reply .github/workflows/ci.yaml:58]: all good. I saw your previous message after I wrote that. Should be fine

**Pull request merged**